src/App.js

```diff
@@ -1,5 +1,107 @@
+const MissionUtils = require('@woowacourse/mission-utils');
+
 class App {
-  play() {}
+  constructor() {}
+
+  getRandomNumber() {
+    this.computer = [];
+    while (this.computer.length < 3) {
+      const NUMBER = MissionUtils.Random.pickNumberInRange(1, 9);
+      if (!this.computer.includes(NUMBER)) {
+        this.computer.push(NUMBER);
+      }
+    }
+    MissionUtils.Console.print('숫자 야구 게임을 시작합니다.');
+  }
+
+  getInputNumber() {
+    MissionUtils.Console.readLine('숫자를 입력해주세요 : ', (number) => {
+      this.checkInputNumber(String(number).split(''));
+    });
+  }
+
+  getBalls(inputNumber) {
+    let ball = 0;
+    inputNumber.forEach((num, idx) => {
+      if (
+        this.computer.includes(Number(num)) &&
+        this.computer.indexOf(Number(num)) !== idx
+      ) {
+        ball++;
+      }
+    });
+    return ball;
+  }
+
+  getStrikes(inputNumber) {
+    let strike = 0;
+    inputNumber.forEach((num, idx) => {
+      if (Number(num) === this.computer[idx]) {
+        strike++;
+      }
+    });
+    return strike;
+  }
+
+  threeStrikes() {
+    MissionUtils.Console.print('3스트라이크');
+    MissionUtils.Console.print('3개의 숫자를 모두 맞히셨습니다! 게임 종료');
+    MissionUtils.Console.readLine(
+      '게임을 새로 시작하려면 1, 종료하려면 2를 입력하세요.',
+      (number) => {
+        if (Number(number) === 1) {
+          this.getRandomNumber();
+          this.getInputNumber();
+        } else if (Number(number) === 2) {
+          MissionUtils.Console.close();
+        } else {
+          throw '입력값이 잘못되었습니다.';
+        }
+      }
+    );
+  }
+
+  getHintMessage(inputNumber) {
+    const BALL = this.getBalls(inputNumber);
+    const STRIKE = this.getStrikes(inputNumber);
+    if (STRIKE === 3) {
+      this.threeStrikes();
+    } else if (BALL && !STRIKE) {
+      MissionUtils.Console.print(`${BALL}볼`);
+      this.getInputNumber();
+    } else if (BALL && STRIKE) {
+      MissionUtils.Console.print(`${BALL}볼 ${STRIKE}스트라이크`);
+      this.getInputNumber();
+    } else if (!BALL && STRIKE) {
+      MissionUtils.Console.print(`${STRIKE}스트라이크`);
+      this.getInputNumber();
+    } else if (!BALL && !STRIKE) {
+      MissionUtils.Console.print('낫싱');
+      this.getInputNumber();
+    }
```

Comment:
`else`를 최대한 안 쓸 수 있다면 안쓰는게 좋아요! 그게 가독성이 높아지기도 하고 js 컨벤션에 포함되기도 하구요!
https://github.com/airbnb/javascript#blocks--no-else-return

다음과 같이 해보는건 어떨까요?
```suggestion
    if (STRIKE === 3) {
      return this.threeStrikes();
    }
    if (BALL && !STRIKE) {
      MissionUtils.Console.print(`${BALL}볼`);
      return this.getInputNumber();
    } 
    if (BALL && STRIKE) {
      MissionUtils.Console.print(`${BALL}볼 ${STRIKE}스트라이크`);
      return this.getInputNumber();
    } 
    if (!BALL && STRIKE) {
      MissionUtils.Console.print(`${STRIKE}스트라이크`);
      return this.getInputNumber();
    } 
    MissionUtils.Console.print('낫싱');
    return this.getInputNumber();
```

Reply:
오오 확실히 else-if를 쓰지 않고 if로만 쓰니 보기 편하네요! 이 부분 코드가 복잡해보여서 수정하고 싶었는데 else-if를 지워도 조건만 맞으면 되니까 문제가 되지 않겠군요. 리뷰 감사합니다 😁